packages/assets-controllers/src/AssetsContractController.test.ts

```diff
@@ -56,31 +61,39 @@ const TEST_ACCOUNT_PUBLIC_ADDRESS =
  */
 async function setupAssetContractControllers({
   options,
-  useNetworkControllerProvider,
+  useNetworkControllerProvider = false,
   infuraProjectId = '341eacb578dd44a1a049cbc5f6fd4035',
 }: {
-  options?: Partial<ConstructorParameters<typeof AssetsContractController>[0]>;
+  options?: Partial<
+    Omit<ConstructorParameters<typeof AssetsContractController>[0], 'messenger'>
+  >;
   useNetworkControllerProvider?: boolean;
   infuraProjectId?: string;
 } = {}) {
   const networkClientConfiguration = {
     type: NetworkClientType.Infura,
-    network: 'mainnet',
+    network: NetworkType.mainnet,
     infuraProjectId,
     chainId: BUILT_IN_NETWORKS.mainnet.chainId,
     ticker: BUILT_IN_NETWORKS.mainnet.ticker,
   } as const;
   let provider: Provider;
 
-  const messenger: NetworkControllerMessenger =
-    new ControllerMessenger().getRestricted({
+  const controllerMessenger = new ControllerMessenger<
+    | NetworkControllerActions
+    | AssetsContractControllerActions
+    | AssetsContractAllowedActions,
+    | NetworkControllerEvents
+    | AssetsContractControllerEvents
+    | AssetsContractAllowedEvents
+  >();
```

Comment:
The function is updated to use a shared, unrestricted controller messenger for communications between NetworkController and AssetsContractController.